src/sentry/services/hybrid_cloud/organization/service.py

```diff
@@ -303,6 +305,20 @@ def send_sso_link_emails(
     ) -> None:
         pass
 
+    @regional_rpc_method(resolve=ByOrganizationId())
+    @abstractmethod
+    def delete_organization(
+        self, *, organization_id: int, user: RpcUser
+    ) -> RpcOrganizationDeleteResponse:
+        pass
+
+    @regional_rpc_method(resolve=ByOrganizationId())
+    @abstractmethod
+    def create_org_delete_log(
+        self, *, organization_id: int, audit_log_actor: RpcAuditLogEntryActor
+    ) -> None:
```

Comment:
If we moved email + audit log generation into `delete_organization` we could skip this as you'd already be in the region for the delete operation.

Reply:
@markstory The audit log generation relies on `request` objects, and I wasn't sure how to pass them around as rpc objects. So, I kept the final audit log generation on the control side.

Reply:
@dashed Not all log generation methods rely on the request object.  There is a `log_service` that will allow you to create the audit logs while passing in all arguments directly.  I think the request value it searches for is IP address, which is tricky.  You could have source IP address be an optional method argument and thread it that way.